### DISCUSSION TITLE
Primes Factorization

### DISCUSSION THREAD

**Discussion by vitaly-t** [vitaly-t/prime-lib#12] [Announcements]
I added initial implementation of [primeFactors](https://github.com/vitaly-t/prime-lib/blob/main/src/prime-factors.ts) function today. It works fine, for most of the practical cases. But it is very basic, and may get stuck on some interesting large numbers, like `2**50 - 2`.

I will work on improving its performance when I get a chance. In the meantime, if somebody wants to contribute - PR-s are welcome!

**Comment by vitaly-t** [comment]
The function has been fully finished, and it is mighty fast :)

**Reply by ishandutta2007** [reply]
@vitaly-t Can you add it to [benchmarks page](https://github.com/vitaly-t/prime-lib/tree/main/benchmarks#readme)

**Reply by vitaly-t** [reply]
If I get a chance to look at it again. In the meantime, I remember looking at a number of implementations in JavaScript, and mine was the fastest one that I could find, i.e. I took the fastest implementation out there, and then optimized it thoroughly.
